server/src/test/java/GroceryFamily/GroceryMom/api/ProductAPITest.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.client.HttpServerErrorException;
+
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class ProductAPITest {
+
+    private MockMvc mockMvc;
+    @InjectMocks
+    private ProductAPI productAPI;
+    @Mock
+    private ProductService service;
+    private List<Product> products;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        this.mockMvc = MockMvcBuilders.standaloneSetup(productAPI).build();
+
+        Set<Price> prices = new HashSet<>();
+        Price price = Price.builder()
+                .unit("pc")
+                .currency("aden")
+                .amount(new BigDecimal(100))
+                .build();
+        prices.add(price);
+        Product product = Product.builder()
+                .namespace("zoo.buy")
+                .code("32133")
+                .name("Tiger 100 kg")
+                .prices(prices)
+                .build();
+        Product product1 = Product.builder()
+                .namespace("zoo.gou")
+                .code("42123")
+                .name("Rock 100 kg")
+                .prices(prices)
+                .build();
+        this.products = new ArrayList<>();
+        products.add(product);
+        products.add(product1);
+
+    }
+
+    @Test
+    void testListProductsByPageSize() throws Exception {
+
+        Page<Product> testPage = Page.<Product>builder()
+                .content(products)
+                .nextPageToken("some_Token")
+                .build();
+
+        when(service.list(anyInt())).thenReturn(testPage);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/products?pageSize=10"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.nextPageToken", is("some_Token")));
+
+    }
+
+    @Test
+    void testListProductsByPageToken() throws Exception {
+
+        Page<Product> testPage = Page.<Product>builder()
+                .content(products)
+                .nextPageToken("some_Token")
+                .build();
+
+        when(service.list(anyString())).thenReturn(testPage);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/products?pageToken=some_token"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.nextPageToken", is("some_Token")))
+                .andExpect(jsonPath("$.content.length()", is(2)));
+    }
+
+    @Test
+    void getProductByIdTest() {
+
+        when(service.get("zoo.buy:32133")).thenReturn(products.get(0));
+
+        Product actualProduct = productAPI.get("zoo.buy:32133");
+
+        Assertions.assertEquals(products.get(0), actualProduct);
```

Comment:
Please use assertions from `assertj` library (check other project's tests for better understanding).